src/validationinterface.cpp

```diff
@@ -104,7 +104,9 @@ void UnregisterAllValidationInterfaces() {
 
 void CMainSignals::MempoolEntryRemoved(CTransactionRef ptx, MemPoolRemovalReason reason) {
     if (reason != MemPoolRemovalReason::BLOCK && reason != MemPoolRemovalReason::CONFLICT) {
-        internals->TransactionRemovedFromMempool(ptx);
+        internals->schedulerClient.AddToProcessQueue([ptx, this] {
```

Comment:
This is being run in a background thread, so there are no possible lock inversions? We've had great success with DEBUG_LOCKORDER and havent had any serious deadlock issues afair since the 0.3.X era.